mp/src/game/server/momentum/mom_fx.cpp

```diff
@@ -0,0 +1,235 @@
+//====== Copyright � 1996-2005, Valve Corporation, All rights reserved. =======
+//
+//
+//
+//=============================================================================
+
+#include "cbase.h"
+#include "mom_fx.h"
+#include "basetempentity.h"
+#include "coordsize.h"
+#include "mom_shareddefs.h"
+#include "weapon/mom_weapon_parse.h"
+
+//=============================================================================
+//
+// Explosions.
+//
+class CTETFExplosion : public CBaseTempEntity
+{
+  public:
+    DECLARE_CLASS(CTETFExplosion, CBaseTempEntity);
+    DECLARE_SERVERCLASS();
+
+    CTETFExplosion(const char *name);
+
+  public:
+    Vector m_vecOrigin;
+    Vector m_vecNormal;
+    CWeaponID m_iWeaponID;
+    int m_nEntIndex;
+};
+
+// Singleton to fire explosion objects
+static CTETFExplosion g_TETFExplosion("TFExplosion");
+
+//-----------------------------------------------------------------------------
+// Purpose:
+// Input  : *name -
+//-----------------------------------------------------------------------------
+CTETFExplosion::CTETFExplosion(const char *name) : CBaseTempEntity(name)
+{
+    m_vecOrigin.Init();
+    m_vecNormal.Init();
+    m_iWeaponID = WEAPON_NONE;
+    m_nEntIndex = 0;
+}
+
+IMPLEMENT_SERVERCLASS_ST(CTETFExplosion, DT_TETFExplosion)
+    SendPropFloat(SENDINFO_NOCHECK(m_vecOrigin[0]), -1, SPROP_COORD_MP_INTEGRAL),
+    SendPropFloat(SENDINFO_NOCHECK(m_vecOrigin[1]), -1, SPROP_COORD_MP_INTEGRAL),
+    SendPropFloat(SENDINFO_NOCHECK(m_vecOrigin[2]), -1, SPROP_COORD_MP_INTEGRAL),
+    SendPropVector(SENDINFO_NOCHECK(m_vecNormal), 6, 0, -1.0f, 1.0f),
+    SendPropInt(SENDINFO_NOCHECK(m_iWeaponID), Q_log2(WEAPON_MAX) + 1, SPROP_UNSIGNED),
+    SendPropInt(SENDINFO_NAME(m_nEntIndex, entindex), MAX_EDICT_BITS), 
+END_SEND_TABLE()
+
+void TE_TFExplosion(IRecipientFilter &filter, float flDelay, const Vector &vecOrigin, const Vector &vecNormal,
+                    CWeaponID iWeaponID, int nEntIndex)
+{
+    VectorCopy(vecOrigin, g_TETFExplosion.m_vecOrigin);
+    VectorCopy(vecNormal, g_TETFExplosion.m_vecNormal);
+    g_TETFExplosion.m_iWeaponID = iWeaponID;
+    g_TETFExplosion.m_nEntIndex = nEntIndex;
+    // Send it over the wire
+    g_TETFExplosion.Create(filter, flDelay);
+}
+
+//=============================================================================
+//
+// TF ParticleEffect
+//
+class CTETFParticleEffect : public CBaseTempEntity
```

Comment:
Why does the particle effect and explosion get separated into their own files on the client but not on the server?

Reply:
It's the way that TF2 originally handled it so I implemented it this way.

Reply:
I would separate them the same way as client, a general `mom_fx` file for two specific TF effects is not favorable. You could probably even make them a shared class files, using `#ifdef`s to separate CLIENT_DLL and an `#else` for `GAME_DLL` code.

Reply:
I added them to fx_mom_shared.cpp instead.